main.tf

```
@@ -97,6 +110,14 @@ resource "aws_mq_broker" "default" {
 
   security_groups = local.broker_security_groups
 
+  dynamic "configuration" {
+    for_each = local.configuration_data_create
```

are you sure you can iterate on bool?
```suggestion
            for_each = local.configuration_data_create
```